nri-redis/test/Spec/Settings.hs

```diff
@@ -0,0 +1,248 @@
+module Spec.Settings (tests) where
+
+import Database.Redis hiding (Ok)
+import qualified Dict
+import qualified Environment
+import qualified Expect
+import qualified Redis.Settings as Settings
+import qualified Test
+import Prelude (Either (..), Show (..))
+
+tests :: Test.Test
+tests =
+  Test.describe "parsing settings from environment"
+    [ decoderTests,
+      decoderWithEnvVarPrefixTests,
+      decoderWithCustomConnectionStringTests
+    ]
+
+-- we lean on `show` for equality since `Database.Redis.ConnectInfo` doesn't have
+-- an `Eq` instance (but does have a `Show` instance)
+
+decoderTests :: Test.Test
+decoderTests =
+  Test.describe "decoder tests"
+    [ Test.test "returns expected defaults when no env vars set" <| \_ ->
+        case parseConnectInfo "redis://localhost:6379" of
+          Left err -> Expect.fail <| "you wrote this test wrong, got err: " ++ Text.fromList err
```

Comment:
All `Left`'s are the same. Would it help to extract those as well so we can clean-up branches on the test itself?